src/execution_plan/ops/op_merge.c

```diff
@@ -180,4 +172,15 @@ OpResult OpMergeReset(OpBase *ctx) {
 }
 
 void OpMergeFree(OpBase *ctx) {
+	OpMerge *op = (OpMerge *)ctx;
+
+	uint node_count = array_len(op->nodes_to_merge);
+	for(uint i = 0; i < node_count; i ++) {
+		PropertyMap_Free(op->nodes_to_merge[i].properties);
```

Comment:
Plugged leak?

Reply:
Yes; this should have been there all along.